lib/common/sanitizer.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+module.exports = SanitizerFactory;
+
+SanitizerFactory.$provide = 'Sanitizer';
+SanitizerFactory.$inject = [
+    'Services.Encryption'
+];
+
+function SanitizerFactory (encryption) {
+    /**
+     * encrypt sensitve information.
+     * @constructor
+     */
+    //function Sanitizer () {
+    //}
+
+    /**
+     *
+     */
+    return function Sanitizer (workflow) {
+        //workflow = JSON.parse(workflow);
+        var sanitizedworkflow = workflow;
+        var walked = [];
+        var stack = [ { workflow : workflow , stack : '' } ];
+        while ( stack.length > 0 ) {
```

Comment:
General question on this function: rather than pushing object properties onto a stack, would it be easier to use a recursive function?  Maybe something like: 
```
function encryptObj(obj) {
    _.forOwn(obj, function(value, key) {
        if (typeof(value) === 'object') {
            encryptObj(value);
        } else if (key.match(secretsPattern) {
            obj[key] = encrypt(value);
        }
    }
}
```